### PR TITLE
tetrio-desktop: init at 8.0.0

### PR DESCRIPTION
**Pull request by wackbyte**
###### Things done

- Built on platform(s)
  - [x] x86_64-linux
  - [ ] aarch64-linux
  - [ ] x86_64-darwin
  - [ ] aarch64-darwin
- [ ] For non-Linux: Is `sandbox = true` set in `nix.conf`? (See [Nix manual](https://nixos.org/manual/nix/stable/command-ref/conf-file.html))
- [ ] Tested, as applicable:
  - [NixOS test(s)](https://nixos.org/manual/nixos/unstable/index.html#sec-nixos-tests) (look inside [nixos/tests](https://github.com/NixOS/nixpkgs/blob/master/nixos/tests))
  - and/or [package tests](https://nixos.org/manual/nixpkgs/unstable/#sec-package-tests)
  - or, for functions and "core" functionality, tests in [lib/tests](https://github.com/NixOS/nixpkgs/blob/master/lib/tests) or [pkgs/test](https://github.com/NixOS/nixpkgs/blob/master/pkgs/test)
  - made sure NixOS tests are [linked](https://nixos.org/manual/nixpkgs/unstable/#ssec-nixos-tests-linking) to the relevant packages
- [x] Tested compilation of all packages that depend on this change using `nix-shell -p nixpkgs-review --run "nixpkgs-review rev HEAD"`. Note: all changes have to be committed, also see [nixpkgs-review usage](https://github.com/Mic92/nixpkgs-review#usage)
- [x] Tested basic functionality of all binary files (usually in `./result/bin/`)
- [21.11 Release Notes (or backporting 21.05 Release notes)](https://github.com/NixOS/nixpkgs/blob/master/CONTRIBUTING.md#generating-2111-release-notes)
  - [ ] (Package updates) Added a release notes entry if the change is major or breaking
  - [ ] (Module updates) Added a release notes entry if the change is significant
  - [ ] (Module addition) Added a release notes entry if adding a new NixOS module
  - [ ] (Release notes changes) Ran `nixos/doc/manual/md-to-db.sh` to update generated release notes
- [ ] Fits [CONTRIBUTING.md](https://github.com/NixOS/nixpkgs/blob/master/CONTRIBUTING.md).

### CODE REVIEW

File: pkgs/games/tetrio-desktop/default.nix [hunk 1]
```diff
@@ -0,0 +1,82 @@
...
+, systemd
+, wrapGAppsHook
+}:
+
+stdenv.mkDerivation rec {
+  pname = "tetrio-desktop";
+  version = "8.0.0";
+
+  src = fetchurl {
+    url = "https://tetr.io/about/desktop/builds/TETR.IO%20Setup.deb";
```
Comment (SuperSandro2000) [inline pkgs/games/tetrio-desktop/default.nix:21]:
```suggestion
    url = "https://web.archive.org/web/20211130172544/https://tetr.io/about/desktop/builds/TETR.IO%20Setup.deb";
```

File: pkgs/games/tetrio-desktop/default.nix [hunk 2]
```diff
@@ -0,0 +1,82 @@
...
+  nativeBuildInputs = [
+    alsa-lib
+    autoPatchelfHook
+    cups
+    libX11
+    libXScrnSaver
+    libXtst
+    mesa
+    nss
+    wrapGAppsHook
```
Comment (SuperSandro2000) [inline pkgs/games/tetrio-desktop/default.nix:35]: Why do we need those libraries in here?

File: pkgs/games/tetrio-desktop/default.nix [hunk 3]
```diff
@@ -0,0 +1,82 @@
...
+    libXtst
+    mesa
+    nss
+    systemd
+  ];
+
+  unpackPhase = ''
+    mkdir -p $TMP/${pname} $out/bin
+    cp $src $TMP/${pname}.deb
+    ar vx $TMP/${pname}.deb
```
Comment (SuperSandro2000) [inline pkgs/games/tetrio-desktop/default.nix:54]: substituting pname a lot makes the code harder to understand.

File: pkgs/games/tetrio-desktop/default.nix (hunk 2 above)
Comment (wackbyte) [reply pkgs/games/tetrio-desktop/default.nix:35]: if I remove them, it fails to build or immediately crashes when I run it

Comment (wackbyte) [reply pkgs/games/tetrio-desktop/default.nix:35]: I'll check

Comment (wackbyte) [reply pkgs/games/tetrio-desktop/default.nix:35]: `autoPatchelfHook` needs them

File: pkgs/games/tetrio-desktop/default.nix (hunk 3 above)
Comment (wackbyte) [reply pkgs/games/tetrio-desktop/default.nix:54]: would you like me to substitute all of them with `tetrio-desktop`?

Comment (SuperSandro2000) [reply pkgs/games/tetrio-desktop/default.nix:57]: I personally find that easier to understand.

Comment (wackbyte) [reply pkgs/games/tetrio-desktop/default.nix]: alright, changed

**Pull request merged**